test/support/url_helpers.rb

```diff
@@ -0,0 +1,19 @@
+# rubocop:disable Lint/NestedMethodDefinition
+
+module UrlHelpers
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def enable_url_helpers
+      # See http://jakegoulding.com/blog/2011/02/26/using-named-routes-in-actionmailer-tests-with-rails-3/
+      include Rails.application.routes.url_helpers
+      Rails.application.routes.default_url_options[:host] = "example.com"
+
+      def default_url_options
+        Rails.application.routes.default_url_options
+      end
```

Comment:
Is this method nested to avoid mixing it into classes that use this module?

Reply:
Yep, wanted to keep it as isolated as it was before I extracted it into a module

Reply:
So is the idea that calling `#default_url_options` in a view context usually calls the Rails built-in method, but if you call `#enable_url_helpers` first it calls the one from `UrlHelpers` ?

Seems a bit odd to dynamically define a method by calling another one. I've not seen this pattern before.